pkg/actuators/machine/instances.go

```diff
@@ -199,24 +202,41 @@ func getBlockDeviceMappings(blockDeviceMappings []providerconfigv1.BlockDeviceMa
 		klog.Errorf("No image for given AMI was found")
 		return nil, fmt.Errorf("no image for given AMI not found")
 	}
-	deviceName := describeAMIResult.Images[0].RootDeviceName
-
-	// Only support one blockDeviceMapping
-	volumeSize := blockDeviceMappings[0].EBS.VolumeSize
-	volumeType := blockDeviceMappings[0].EBS.VolumeType
-	blockDeviceMapping := ec2.BlockDeviceMapping{
-		DeviceName: deviceName,
-		Ebs: &ec2.EbsBlockDevice{
-			VolumeSize: volumeSize,
-			VolumeType: volumeType,
-			Encrypted:  blockDeviceMappings[0].EBS.Encrypted,
-		},
-	}
-	if *volumeType == "io1" {
-		blockDeviceMapping.Ebs.Iops = blockDeviceMappings[0].EBS.Iops
+
+	onlyOneEmptyDeviceName := false
+	for _, blockDeviceMappingSpec := range blockDeviceMappingSpecs {
+		if blockDeviceMappingSpec.EBS == nil {
+			continue
+		}
+
+		deviceName := blockDeviceMappingSpec.DeviceName
+		volumeSize := blockDeviceMappingSpec.EBS.VolumeSize
+		volumeType := blockDeviceMappingSpec.EBS.VolumeType
+
+		if blockDeviceMappingSpec.DeviceName == nil {
+			if onlyOneEmptyDeviceName {
```

Comment:
I think this name could be a little confusing maybe, since we go into this if statement if there is more than one empty device name. Perhaps `rootDeviceFound` could be a better name?

Reply:
👍